voir/instruments/cpu.py

```diff
@@ -16,3 +17,59 @@ def monitor():
         }
 
     return monitor
+
+
+def retrieve_process_info(process, acc):
+    io = process.io_counters()
+    acc["read_bytes"] += io.read_bytes
+    acc["write_bytes"] += io.write_bytes
+    acc["read_chars"] += io.read_chars
+    acc["write_chars"] += io.write_chars
+
+    mem = process.memory_info()
+    acc["mem_used"] += mem.rss
+
+    cpu = process.cpu_percent(interval=1)
+    acc["cpu_percent"] += cpu
+    acc["children"] += 1
+
+
+def _recursive(process, acc):
+    for child in process.children(recursive=True):
+        try:
+            retrieve_process_info(child, acc)
+        except psutil.AccessDenied:
+            pass
+
+
+def process_monitor(pid, recursive=True):
+    process = psutil.Process(pid)
+    cpu_count = psutil.cpu_count()
+    mem_total = psutil.virtual_memory()
+
+    def monitor():
+        nonlocal process
+        acc = defaultdict(float)
+
+        with process.oneshot():
+            cpu_num = process.cpu_num()
+            retrieve_process_info(process, acc)
```

Comment:
Should we also catch `psutil.AccessDenied` here?

Reply:
done